Move scripts from nblibrary to cupid_utils/

mnlevy1981 opened the pull request:
### Description of changes:

Created a new python package named cupid_utils that has a license, README, `pyproject.toml`, `__init__.py` files, and the actual .py modules we had been keeping in `nblibrary/<component>/`. This serves two purposes:

1. Users should be able to re-run notebooks directly in the computed_notebooks directory without needing to copy any files in
2. Having all these modules co-located will hopefully make it easier to look closer at what each one does and see if it actually belongs in CUPiD

Note that users will have to update (cupid-analysis) either by running pip install -e cupid_utils or by rerunning the conda install command.

#### All PRs Checklist:
* [x] Have you followed the guidelines in our [Contributor's Guide](https://ncar.github.io/CUPiD/contributors_guide.html)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you made sure that the [`pre-commit` checks passed (#8 in Adding Notebooks Guide)](https://ncar.github.io/CUPiD/addingnotebookstocollection.html)?
* [x] Have you successfully tested your changes locally when running standalone CUPiD?
* [x] Once you are ready to have your PR reviewed, have you changed it from a Draft PR to an Open PR?

mnlevy1981 committed:
Move scripts from nblibrary to cupid_utils/

Created a new python package named cupid_utils that has a license, README,
pyproject.toml, __init__.py files, and the actual .py modules we had been
keeping in nblibrary/<component>/. This serves two purposes:

1. Users should be able to re-run notebooks directly in the computed_notebooks
directory without needing to copy any files in
2. Having all these modules co-located will hopefully make it easier to look
closer at what each one does and see if it actually belongs in CUPiD

Note that users will have to update (cupid-analysis) either by running pip
install -e cupid_utils or by rerunning the conda install command.

mnlevy1981 committed:
Merge branch 'main' into separate_utils

mnlevy1981 committed:
Cleanup for ROF notebooks

1. ROF notebooks failed because of issue using dict += construct
2. ROF notebooks failed because config.yml was using h0a instead of h0
3. ROF notebooks failed because I hadn't moved setup/ to cupid_utils

TeaganKing reviewed (approved):
Looks great! Thanks for fixing that rof error, too!
I think we just need to move the yaml/config files out of nblibrary still as well as test in CESM workflow.

mnlevy1981 committed:
Merge branch 'main' into separate_utils

mnlevy1981 committed:
move regional_utils.py to cupid_utils/ocn

mnlevy1981 committed:
Move sea ice YAML files to cupid_utils/input_files

mnlevy1981 commented:
There are two non-notebook files left in `nblibrary/`:

```
$ find nblibrary/ -type f | grep -v ipynb
nblibrary/atm/config_f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae.001.yaml
nblibrary/lnd/model_setup.txt
```

I think these might have both been committed by mistake? The first is an ADF config file, and the second is an ILAMB input file... I'm tempted to just `git rm` them and then see if anything fails in our testing.

TeaganKing committed:
revert case_dic change because it's now in Naoki's PR

TeaganKing committed:
revert #2

TeaganKing committed:
revert rof ocean nb

TeaganKing committed:
revert #3

mnlevy1981 committed:
Merge branch 'main' into separate_utils

mnlevy1981 committed:
Lots of clean-up for runoff

1. Output files are written to proper directory rather than nblibrary/rof
2. Don't call client.shutdown() from CUPiD
3. Better logic for separating base case and coming up with analysis_name

mnlevy1981 committed:
Merge remote-tracking branch 'origin/separate_utils' into separate_utils
